cfme/infrastructure/host.py

```diff
@@ -234,6 +234,31 @@ def power_off(self):
         pow_btn('Power Off', invokes_alert=True)
         sel.handle_alert()
 
+    def get_power_state(self):
+        return self.get_detail("Properties", "Power State")
+
+    def refresh(self, cancel=False):
+        tb.select("Configuration", "Refresh Relationships and Power States", invokes_alert=True)
+        sel.handle_alert(cancel=cancel)
+
+    def wait_for_host_state_change(self, desired_state, timeout=300, provider_crud=None):
```

Comment:
**REQUIRED: ** I'll allow this for now, but the Host object should really have a provider object associated with it. This is not your fault, and there are situations where it won't have one, but in this case, there should be one. Please can you raise an Issue requesting that we add it to the Host object and put a TODO here in the code that provider_crud shouldn't be necessary and that it should pull the provider from the ```self.provider```.